ci/local/build.sh

```diff
@@ -67,38 +67,71 @@ BASE_CONTAINER_BUILD_DIR=${REPO_PATH}/build_$(echo $(basename ${DOCKER_IMAGE})|s
 CPP_CONTAINER_BUILD_DIR=${BASE_CONTAINER_BUILD_DIR}/cpp
 PYTHON_CONTAINER_BUILD_DIR=${BASE_CONTAINER_BUILD_DIR}/python
 
-
+# Note the prevention of expanding variables. Preventing expansion is critical as
+# expansion in context of calling shell would lead to incorrect result as not
+# all variables are set
 BUILD_SCRIPT="#!/bin/bash
 set -e
 WORKSPACE=${REPO_PATH_IN_CONTAINER}
 PREBUILD_SCRIPT=${REPO_PATH_IN_CONTAINER}/ci/gpu/prebuild.sh
 BUILD_SCRIPT=${REPO_PATH_IN_CONTAINER}/ci/gpu/build.sh
-cd ${WORKSPACE}
+cd \${WORKSPACE}
 if [ -f \${PREBUILD_SCRIPT} ]; then
     source \${PREBUILD_SCRIPT}
 fi
 yes | source \${BUILD_SCRIPT}
 "
 
-if (( ${CONTAINER_SHELL_ONLY} == 0 )); then
+if (( CONTAINER_SHELL_ONLY == 0 )); then
```

Comment:
This is done because of: https://github.com/koalaman/shellcheck/wiki/SC2004 $/${} is unnecessary on arithmetic variables.